libp2p-swarm/src/transport.rs

```diff
@@ -307,19 +307,19 @@ pub enum EitherListenStream<A, B> {
 
 impl<A, B, Sa, Sb> Stream for EitherListenStream<A, B>
 where
-	A: Stream<Item = (Sa, Multiaddr), Error = IoError>,
-	B: Stream<Item = (Sb, Multiaddr), Error = IoError>,
+	A: Stream<Item = (Result<Sa, IoError>, Multiaddr), Error = IoError>,
```

Comment:
In general if type parameters get longer than 1 character I tend to give them real names (like in this case I'd call them `AStream` and `BStream`).